code/game/gamemodes/changeling/powers/transform.dm

```diff
@@ -11,36 +11,93 @@
 	name = "flesh"
 	flags = NODROP
 
+/obj/item/clothing/glasses/changeling/attack_hand(mob/user)
+	if(loc == user && user.mind && user.mind.changeling)

```

Comment:
Ah, i misunderstood. Assumed you wanted to just leave this as flags only. Yeah that would work.